ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/rest/RestTestExecutor.java

```diff
@@ -319,6 +348,57 @@ private static StatementSplit splitStatements(final RestTestCase testCase) {
     return Optional.of(resp.getResponse());
   }
 
+  private Optional<List<StreamedRow>> sendQueryStatement(
+      final RestTestCase testCase,
+      final String sql,
+      final Runnable afterHeader
+  ) {
+    final RestResponse<StreamPublisher<StreamedRow>> resp
+        = restClient.makeQueryRequestStreamed(sql, null);
+
+    if (resp.isErroneous()) {
+      handleErrorResponse(testCase, resp);
+      return Optional.empty();
+    }
+
+    return handleRowPublisher(resp.getResponse(), afterHeader);
+  }
+
+  private Optional<List<StreamedRow>> handleRowPublisher(
+      final StreamPublisher<StreamedRow> publisher,
+      final Runnable afterHeader
+  ) {
+    final CompletableFuture<List<StreamedRow>> future = new CompletableFuture<>();
+    final CompletableFuture<StreamedRow> header = new CompletableFuture<>();
+    final QueryStreamSubscriber subscriber = new QueryStreamSubscriber(publisher.getContext(),
+        future, header);
+    publisher.subscribe(subscriber);
+
+    try {
+      header.get();
+    } catch (Exception e) {
+      LOG.error("Error awaiting header", e);
+      throw new RuntimeException(e);
+    }
+
+    try {
+      afterHeader.run();
+    } catch (Exception e) {
+      LOG.error("Error doing after header", e);
```

Comment:
Nope, this was an oversight.  I collapse these three try blocks into one since there's no good reason they should be separate.  I think I had done it to have a slightly better error message, but it should be obvious from the stacktrace.